packit_service/worker/build/build_helper.py

```diff
@@ -34,26 +34,55 @@
 from packit_service import sentry_integration
 from packit_service.config import ServiceConfig, Deployment
 from packit_service.models import SRPMBuildModel
-from packit_service.service.events import (
-    PullRequestGithubEvent,
-    PullRequestCommentGithubEvent,
-    CoprBuildEvent,
-    PushGitHubEvent,
-    ReleaseEvent,
-    PullRequestPagureEvent,
-    PullRequestCommentPagureEvent,
-    KojiBuildEvent,
-)
 from packit_service.trigger_mapping import (
     is_trigger_matching_job_config,
     are_job_types_same,
 )
 from packit_service.worker.reporting import StatusReporter
+from packit_service.service.events import TheJobTriggerType
 from sandcastle import SandcastleTimeoutReached
 
 logger = logging.getLogger(__name__)
 
 
+class BuildHelperMetadata:
+    """
+    Class to represent the data from event needed by BaseBuildJobHelper.
+    """
+
+    def __init__(
+        self,
+        trigger: TheJobTriggerType,
+        pr_id: Optional[int] = None,
+        git_ref: Optional[str] = None,
+        commit_sha: Optional[str] = None,
+        # identifier of the respective event (PR, release) used e.g. in copr project name
+        identifier: Optional[str] = None,
+    ):
+        self.trigger = trigger
+        self.pr_id = pr_id
+        self.git_ref = git_ref
+        self.commit_sha = commit_sha
+        self.identifier = identifier
```

Comment:
It looks like all of these can be calculated from the `db_trigger` object so:
- Name can be something related to trigger.
- We can (?in the future) potentially use the id of the database object only. But I am not sure about this now.

Reply:
can I rename it for now to something like `TriggerMetadata` and later (in some next PR) try to use only DB objects?

Reply:
Yes, can be. I am even not sure now if we want to forward that info via database. It's for another discussion.